app/src/main/AndroidManifest.xml

```diff
@@ -10,16 +10,32 @@
         android:label="@string/app_name"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name=".MainActivity" android:launchMode="singleTop">
```

Comment:
There are 3 scenarios I can think of in respect to deep-linking:

1. The app is closed - in this case of deep-link the user will reach directly to `MovieDetailsLoaderActivity` which after finish loading will later add `MainActivity` and `MovieDetailsActivity` to the back stack.
2. The app is open in the main page (`MainActivity`) - In case of deep-link then I assume that `singleTop` will on `MainActivity` will prevent having two instances of it in the back stack.
3. The app is open in a movie details page (or any page of the app that is not main) - In this case, new instances of `MainActivity` and `MovieDetailsActivity` will be launched, which mean that tapping back from `MainActivity` will lead the user to the previous movie details page that was shown before the deep-link which is probably not what you want (you want that tapping back from the main page will always close the app).